src/languageservice/services/yamlSchemaService.ts

```diff
@@ -152,9 +161,19 @@ export class YAMLSchemaService extends JSONSchemaService {
     dependencies: SchemaDependencies
   ): Promise<ResolvedSchema> {
     const resolveErrors: string[] = schemaToResolve.errors.slice(0);
-    let schema = schemaToResolve.schema;
+    let schema: JSONSchema = schemaToResolve.schema;
     const contextService = this.contextService;
 
+    if (!schema07Validator(schema)) {
+      const errs: string[] = [];
+      for (const err of schema07Validator.errors as DefinedError[]) {
+        errs.push(`${err.instancePath} : ${err.message}`);
+      }
+      resolveErrors.push(
+        `Schema '${schemaToResolve.schema.title ?? schemaToResolve.schema.url}' is not valid:\n${errs.join('\n')}`
```

Comment:
This is not sufficient to find a friendly name for the schema. When I tried this with a schema referenced locally with a modeline the schema is resolved as `undefined`. Can we use the same mechanics that is used on the Code lens for Jump to schema code lens.